### PR TITLE
add lychrel number sequence

### PR DESCRIPTION
**Pull request by t-m-z**

### CODE REVIEW

File: assets/i18n/en.json [hunk 1]
```diff
@@ -2446,6 +2446,9 @@
   "numbersequence_recaman_title": "Recamán Sequence",
   "numbersequence_recaman_description": "OEIS A005132, until n = 11111",
   "numbersequence_recaman_example": "0, 1, 3, 6, 2, 7, 13, 20, 12, 21, 11, 22, 10, 23, 9, 24, ...",
+  "numbersequence_lychrel_title": "Lychrel Sequence",
+  "numbersequence_lychrel_description": "OEIS A023108, until n = 10000",
```
Comment (S-Man42) [inline assets/i18n/en.json:2450]: n defines not the max number of the sequences but the n-th number. See Mersenne Primes: n = 18. It supports only 18 Numbers. :)

File: lib/widgets/tools/science_and_technology/number_sequences/lychrel.dart [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import 'package:gc_wizard/logic/tools/science_and_technology/number_sequence.dart';
+import 'package:gc_wizard/widgets/tools/science_and_technology/number_sequences/base/numbersequences_check.dart';
+import 'package:gc_wizard/widgets/tools/science_and_technology/number_sequences/base/numbersequences_containsdigits.dart';
+import 'package:gc_wizard/widgets/tools/science_and_technology/number_sequences/base/numbersequences_digits.dart';
+import 'package:gc_wizard/widgets/tools/science_and_technology/number_sequences/base/numbersequences_nthnumber.dart';
+import 'package:gc_wizard/widgets/tools/science_and_technology/number_sequences/base/numbersequences_range.dart';
+
+class NumberSequenceLychrelCheckNumber extends NumberSequenceCheckNumber {
+  NumberSequenceLychrelCheckNumber() : super(mode: NumberSequencesMode.LYCHREL, maxIndex: 555);
```
Comment (S-Man42) [inline lib/widgets/tools/science_and_technology/number_sequences/lychrel.dart:9]: You dont have 555 numbers in the sequence

File: lib/widgets/tools/science_and_technology/number_sequences/lychrel.dart [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+class NumberSequenceLychrelDigits extends NumberSequenceDigits {
+  NumberSequenceLychrelDigits() : super(mode: NumberSequencesMode.LYCHREL, maxDigits: 1111);
```
Comment (S-Man42) [inline lib/widgets/tools/science_and_technology/number_sequences/lychrel.dart:13]: You have no numbers with length = 1111 in the sequence (as far as I see, it's max 4, isnt it?)

File: assets/i18n/en.json (hunk 1 above)
Comment (t-m-z) [reply assets/i18n/en.json:2450]: check

File: lib/widgets/tools/science_and_technology/number_sequences/lychrel.dart (hunk 2 above)
Comment (t-m-z) [reply lib/widgets/tools/science_and_technology/number_sequences/lychrel.dart:9]: check

File: lib/widgets/tools/science_and_technology/number_sequences/lychrel.dart (hunk 3 above)
Comment (t-m-z) [reply lib/widgets/tools/science_and_technology/number_sequences/lychrel.dart:13]: check

**Pull request merged**